codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/KotlinJmespathExpressionVisitor.kt

```diff
@@ -244,12 +243,44 @@ class KotlinJmespathExpressionVisitor(
     }
 
     override fun visitMultiSelectHash(expression: MultiSelectHashExpression): VisitedExpression {
-        throw CodegenException("MultiSelectHashExpression is unsupported")
+        val attributes = StringBuilder()
+        var i = 0
+        expression.expressions.forEach {
+            if (i == expression.expressions.size - 1) {
+                attributes.append("val ${it.key}: T")
+            } else {
+                attributes.append("val ${it.key}: T, ")
+            }
+            i++
+        }
```

Comment:
Kotlin has a [`joinToString` function](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/join-to-string.html) which handles this nicely. The default is a comma-separated string so you don't need to customize the parameters, just the string transformation.

```kt
val attributes = expression.expressions.joinToString {
	"val ${it.key}: T" 
}
```

Reply:
Thanks, I wish I knew about this earlier